### PR TITLE
integrated camera-vision BQ

### PR DESCRIPTION
**Pull request by IJDykeman**

### CODE REVIEW

File: camera/integrated_camera_vision_balsaq.cc [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+#include "vision/fiducial_detection_message.hh"
+
+#include <cstddef>
+#include <iostream>
+#include <sstream>
+#include <thread>
+
+namespace jet {
+
+void CameraBq::init(const Config& config) {
```
Comment (jpanikulam) [inline camera/integrated_camera_vision_balsaq.cc]: Name class same as file

File: camera/integrated_camera_vision_balsaq.cc [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    const Calibration camera_calibration = camera_manager_.get_calibration(message.camera_serial_number);
+    const cv::Mat camera_frame = get_image_mat(message);
+    std::optional<FiducialDetectionMessage> detection_message =
+        create_fiducial_detection_message(camera_frame, camera_calibration, message.header.timestamp_ns);
+    if (detection_message) {
+      fiducial_detection_publisher_->publish(*detection_message);
+    }
+  }
+  if (last_msg_recvd_timestamp_ < get_current_time() - Duration::from_seconds(1)) {
```
Comment (jpanikulam) [inline camera/integrated_camera_vision_balsaq.cc]: Consider parenthesizing the arithmetic

File: camera/integrated_camera_vision_balsaq.cc [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    gonogo().go();
+    last_msg_recvd_timestamp_ = cap_time_vehicle;
+    CameraImageMessage message = create_camera_image_message(camera_frame, camera_.serial_number, cap_time_vehicle);
+
+    image_publisher_->publish(message);
+    std::cout << "Camera BQ: publishes a camera frame " << message.width << " " << message.height << std::endl;
+
+    const Calibration camera_calibration = camera_manager_.get_calibration(message.camera_serial_number);
+    const cv::Mat camera_frame = get_image_mat(message);
+    std::optional<FiducialDetectionMessage> detection_message =
```
Comment (jpanikulam) [inline camera/integrated_camera_vision_balsaq.cc]: `const`

File: camera/integrated_camera_vision_balsaq.cc [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+#include "infrastructure/balsa_queue/bq_main_macro.hh"
+#include "infrastructure/time/duration.hh"
+#include "infrastructure/time/time_utils.hh"
+#include "vision/fiducial_detection_and_pose.hh"
+#include "vision/fiducial_detection_message.hh"
+
+#include <cstddef>
+#include <iostream>
+#include <sstream>
+#include <thread>
```
Comment (jpanikulam) [inline camera/integrated_camera_vision_balsaq.cc]: Looks like this is unused

File: camera/integrated_camera_vision_balsaq.cc [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+#include "config/fiducial_map/read_fiducial_map.hh"
+#include "infrastructure/balsa_queue/bq_main_macro.hh"
+#include "infrastructure/time/duration.hh"
+#include "infrastructure/time/time_utils.hh"
+#include "vision/fiducial_detection_and_pose.hh"
+#include "vision/fiducial_detection_message.hh"
+
+#include <cstddef>
+#include <iostream>
+#include <sstream>
```
Comment (jpanikulam) [inline camera/integrated_camera_vision_balsaq.cc]: This looks unused

File: camera/integrated_camera_vision_balsaq.hh [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+namespace jet {
+
+// A C++ form of the yaml config
+//
+// Stored this way to avoid parsing overhead when re-reading the
+// config each frame
+//
+// The fields are left intentionally uninitialized, so that
+// their use before initialization can be detected with Valgrind
```
Comment (jpanikulam) [inline camera/integrated_camera_vision_balsaq.hh]: These comments are stale

File: camera/integrated_camera_vision_balsaq.hh [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  // associated with that monotonic time
+  static const int CAMERA_FPS = 10;
+  PublisherPtr image_publisher_;
+  PublisherPtr fiducial_detection_publisher_;
+  cv::VideoCapture cap_;
+  Timestamp last_msg_recvd_timestamp_;
+
+  Camera camera_;
+  CameraConfiguration camera_config_;
+  CameraManager camera_manager_ = CameraManager();
```
Comment (jpanikulam) [inline camera/integrated_camera_vision_balsaq.hh]: [nit] `CameraManager camera_manager_;` calls the default constructor anyway

File: camera/integrated_camera_vision_balsaq.cc [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+  std::cout << "Camera BQ: trying to get a frame" << std::endl;
+
+  if (cap_.read(camera_frame)) {
+    // Get the image capture timestamp, which is the time that the first byte was captured, as returned by
+    // clock_gettime().
+    const long int cap_time_msec = cap_.get(cv::CAP_PROP_POS_MSEC);
+    const Timestamp cap_time_vehicle = time::msec_monotonic_to_vehicle_monotonic(cap_time_msec);
+    gonogo().go();
+    last_msg_recvd_timestamp_ = cap_time_vehicle;
+    CameraImageMessage message = create_camera_image_message(camera_frame, camera_.serial_number, cap_time_vehicle);
```
Comment (jpanikulam) [inline camera/integrated_camera_vision_balsaq.cc]: `const`

**Comment by benkroop** [comment]
@IJDykeman have you addressed Jake's nits? If yes, let's get this merged.

**Pull request merged**